File: src/main/java/employeetech/com/employee/manager/model/Payroll.java
package employeetech.com.employee.manager.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;
import java.time.LocalDate;

@Entity
@Table(name = "payrolls")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Payroll implements Serializable {
    @Id
    @GeneratedValue
    private Long id;

    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "employee_id", nullable = false)
    private Employee employee;

    @Column(nullable = false, updatable = false)
    private Double salary;

    private LocalDate payDate;
}

File: src/main/kotlin/employeetech/com/employee/manager/model/Payroll.kt
package employeetech.com.employee.manager.model

import jakarta.persistence.*
import java.io.Serializable
import java.time.LocalDate

@Entity
@Table(name = "payrolls")
data class Payroll(
    @Id
    @GeneratedValue
    val id: Long? = null,

    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    @JoinColumn(name = "employee_id", nullable = false)
    val employee: Employee? = null,

    @Column(nullable = false, updatable = false)
    val salary: Double? = null,

    val payDate: LocalDate? = null
) : Serializable
